tests/MongoDB.Driver.Tests/Specifications/server-discovery-and-monitoring/ServerDiscoveryAndMonitoringProseTests.cs

```diff
@@ -0,0 +1,157 @@
+                _ = client
+                    .GetDatabase(DriverTestConfiguration.DatabaseNamespace.DatabaseName)
+                    .GetCollection<BsonDocument>(DriverTestConfiguration.CollectionNamespace.CollectionName)
+                    .Find(FilterDefinition<BsonDocument>.Empty);
+
+                // Sleep for 2 seconds. This must be long enough for multiple heartbeats to succeed.
+                Thread.Sleep(TimeSpan.FromSeconds(2));
+
+                foreach (ServerDescriptionChangedEvent @event in eventCapturer.Events.ToList())
+                {
+                    @event.NewDescription.HeartbeatException.Should().BeNull();
+                    @event.NewDescription.AverageRoundTripTime.Should().NotBe(default);
+                }
+
+                var failPointCommand = BsonDocument.Parse(
+                    @"{
+                        configureFailPoint : 'failCommand',
+                        mode : { times : 1000 },
+                        data :
+                        {
+                            failCommands : [ 'isMaster' ],
+                            blockConnection : true,
+                            blockTimeMS : 500,
+                            appName : 'streamingRttTest'
+                        }
+                    }");
+
+                using (FailPoint.Configure(client.Cluster, NoCoreSession.NewHandle(), failPointCommand))
+                {
+                    // Note that the Server Description Equality rule means that ServerDescriptionChangedEvents will not be published.
+                    // So we use reflection to obtain the latest RTT instead.
+                    var server = client.Cluster.SelectServer(WritableServerSelector.Instance, CancellationToken.None);
+                    var roundTripTimeMonitor = server._monitor()._roundTripTimeMonitor();
+                    var expectedRoundTripTime = TimeSpan.FromMilliseconds(250);
+                    var timeout = TimeSpan.FromSeconds(30); // should not be reached without a driver bug
+                    SpinWait.SpinUntil(() => roundTripTimeMonitor.Average >= expectedRoundTripTime, timeout).Should().BeTrue();
+                }
+            }
+        }
+
+        // private methods
+        private DisposableMongoClient CreateClient(EventCapturer eventCapturer, TimeSpan heartbeatInterval, string applicationName = null)
+        {
+            var clonedClient = DriverTestConfiguration.Client.Settings.Clone();
+            return DriverTestConfiguration.CreateDisposableClient(
+                (clientSettings) =>
+                {
+                    clientSettings.ApplicationName = applicationName;
+                    clientSettings.HeartbeatInterval = heartbeatInterval;
+                    clientSettings.ClusterConfigurator = builder => builder.Subscribe(eventCapturer);
+                });
+        }
+    }
+
+    internal static class ServerRelfector
```

Comment:
Relfector => Reflector

Usually an XyzReflector lives in a file called XyzTests.